### PR TITLE
fix: return data about already existing users after verifying `phone` and `email`

### PR DESCRIPTION
**Pull request by Danipulok**
**Current behavior:**
User receives data if there are any existing users with specified email and\or phone without verifying their data first.

**PR behavior:**
User receives information about already existing username\email\phone only after verifying their data.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@dacongda plz review

File: object/check.go [hunk 1]
```diff
@@ -80,9 +70,7 @@ func CheckUserSignup(application *Application, organization *Organization, authF
 				return i18n.Translate(lang, "check:Email cannot be empty")
 			}
 		} else {
-			if HasUserByField(organization.Name, "email", authForm.Email) {
-				return i18n.Translate(lang, "check:Email already exists")
-			} else if !util.IsEmailValid(authForm.Email) {
+			if !util.IsEmailValid(authForm.Email) {
```
Comment (dacongda) [inline object/check.go:73]: we will check whether Email item is visible in signupitem firsrt currently , but in the func `CheckExistingUserDetails` you haven't check it, will it make more problem?

**Review by dacongda — APPROVED** [review]
lgtm

File: object/check.go [hunk 2]
```diff
@@ -152,6 +138,19 @@ func CheckUserSignup(application *Application, organization *Organization, authF
 	return ""
 }
 
+func CheckExistingUserDetails(organization *Organization, authForm *form.AuthForm, lang string) string {
+	if HasUserByField(organization.Name, "name", authForm.Username) {
+		return i18n.Translate(lang, "check:Username already exists")
+	}
+	if HasUserByField(organization.Name, "email", authForm.Email) {
```
Comment (hsluoyz) [inline object/check.go:145]: Empty not checked

**Comment by hsluoyz** [comment]
Closed for low code quality

**Comment by Danipulok** [comment]
@halozhy , does it mean that currently there's a low code currently in casdoor?

image

**Pull request closed without merging**